eZ/Publish/Core/Repository/LocationService.php

```diff
@@ -478,7 +478,7 @@ public function updateLocation(APILocation $location, LocationUpdateStruct $loca
         if ($locationUpdateStruct->remoteId !== null) {
             try {
                 $existingLocation = $this->loadLocationByRemoteId($locationUpdateStruct->remoteId);
-                if ($existingLocation !== null) {
+                if ($existingLocation !== null && $loadedLocation->remoteId !== $locationUpdateStruct->remoteId) {
```

Comment:
Because we're throwing the exception if the locations (and thus their IDs) differ.